### ISSUE TITLE
`SIM117` incorrect believes you can merge `async with` and `with`

### ISSUE THREAD

**Issue by thomasdesr** [charliermarsh/ruff#1902]
SIM117 should:
* Not trigger when nesting a context manager within an async context manager since you can't generally merge an async context manager with a sync context manager.
* Should trigger when nesting two async context managers (edit: actually maybe should be a separate lint?)

Example of permutations and their results as of `0.0.222`:

```python
import asyncio
from contextlib import asynccontextmanager, contextmanager


@asynccontextmanager
async def async_ctx_mgr():
    yield "async"


@contextmanager
def sync_ctx_mgr():
    yield "sync"


async def main():
    # True Positive, correctly suggests to minimize
    with sync_ctx_mgr() as first:
        with sync_ctx_mgr() as second:
            print("sync sync", first, second)

    # False positive, incorrectly suggests to minimize
    async with async_ctx_mgr() as first:
        with sync_ctx_mgr() as second:
            print("async sync", first, second)

    # True negative, correctly doesn't flag as an issue
    with sync_ctx_mgr() as first:
        async with async_ctx_mgr() as second:
            print("sync async", first, second)

    # False negative, incorrectly fails to suggest to minimize
    async with async_ctx_mgr() as first:
        async with async_ctx_mgr() as second:
            print("async async", first, second)


if __name__ == "__main__":
    asyncio.run(main())
```

Output from `ruff` on ^
```
$ ruff --isolated --select SIM ./SIM117-repro.py
SIM117-repro.py:16:5: SIM117 Use a single `with` statement with multiple contexts instead of nested `with` statements
SIM117-repro.py:20:5: SIM117 Use a single `with` statement with multiple contexts instead of nested `with` statements
Found 2 error(s).
```

```
$ ruff --version
ruff 0.0.222
```

Note: `flake8-simplify` seems to do the right thing here even though their tests are missing coverage of this case.

Thanks as always!

**Comment by charliermarsh** [comment]
Good call.

**Comment by thomasdesr** [comment]
So fast!

Re-ran with a main build and the sync context manager case looks fixed. Thanks ❤️

**Comment by charliermarsh** [comment]
Awesome :) I opted to just turn it off entirely in the async case. I think it's safe to combine multiple async context managers, but honestly I've done so little async Python that I'd have to check.

**Comment by charliermarsh** [comment]
(Also I'm going to cut a new build tonight, so this will be out in a bit.)

**Comment by thomasdesr** [comment]
Sounds good! Thanks so much :D

I also believe its fine to merge async context managers, but double checking sgtm xD
